compiler/rustc_codegen_llvm/src/type_of.rs

```diff
@@ -396,8 +393,19 @@ impl<'tcx> LayoutLlvmExt<'tcx> for TyAndLayout<'tcx> {
             // up suppressing vectorization as it introduces shifts when it
             // extracts all the individual values.
 
-            let ety = element.llvm_type(cx);
-            return Some(cx.type_vector(ety, *count));
+            if *count <= 2 {
+                // For short arrays, use LLVM's array type which it will unpack
+                // out in optimizations to a scalar or pair of scalars.
+                // (Having types like `<1 x u8>` is silly.)
+                let ety = element.llvm_type(cx);
+                return Some(cx.type_array(ety, *count));
+            } else if count.is_power_of_two() {
```

Comment:
This is already inside `if count.is_power_of_two()`, did you mean to remove it from the outer condition?

Reply:
Good point!  I think I'd intended to remove it from the outer, but with the tests not actually failing I guess I never did.

@rustbot author